test/e2e/kustomizations_test.go

```diff
@@ -119,3 +123,28 @@ func patchLivenessProbes(baseManifestReader io.Reader, deployment k8stypes.Names
 	}
 	return kubectl.GetKustomizedManifest(kustomization, baseManifestReader)
 }
+
+// patchReadinessProbePath patches the given deployment's path of readiness probe.
+// Kong gateway supports endpoint `/status/ready` since 3.4, and prior versions uses `/status`,
+// so we need to change the path of readiness probe of Kong gateway deployment when its version is < 3.4.
+func patchReadinessProbePath(baseManifestReader io.Reader, deployment k8stypes.NamespacedName, probePath string) (io.Reader, error) {
+	kustomization := types.Kustomization{
+		Patches: []types.Patch{
+			{
+				Patch: fmt.Sprintf(readinessProbePathPatch, probePath),
```

Comment:
Just a small suggestion: how about we extract this (and the `const` that it uses) into a function that's responsible for preparing the patch, which take the `probePath` as an argument? Less chance for error (e.g. not providing the format string param for `readinessProbePathPatch` etc.)

Reply:
What do you mean by "extract into a function"? like 
```golang
func prepareReadinessProbePatch(baseManifestReader io.Reader, deployment k8stypes.NamespacedName) func (string) (io.Reader, error) {
// returns the function to patch the readiness probe path of the input manifest reader
}
```
Then, in places where we want to run the patch 
```golang
    patch := prepareReadinessProbePatch(manifest, deployment)
    manifest, err := patch("/status")
```
?

I think it does not bring much effect on using the patch easier and reduce chance for error. This still involve the same 3 params, but one more function call, and one more anonymous function.